src/app/view/components/login-as/login-as.spec.ts

```diff
@@ -114,18 +115,39 @@ describe('LoginAsComponent', () => {
         expect(paginatorService.filter).toEqual('new filter');
     });
 
-    it('should call redirect to the first porlet when login as happen', () => {
-        /*
-            TODO: need to revisit this component, too much going on to init
-        */
-        comp.visible = true;
-        comp.ngOnInit();
-        fixture.detectChanges();
-        comp.form.get('loginAsUser').setValue(mockUser);
-        fixture.detectChanges();
+    it(
+        'should call redirect to the first porlet when login as happen and same porlet is being loaded',
+        fakeAsync(() => {
+            comp.visible = true;
+            comp.ngOnInit();
+            fixture.detectChanges();
```

Comment:
really we need call ngOnInit and detectChanges, detectChanges call ngOnInit automaticly